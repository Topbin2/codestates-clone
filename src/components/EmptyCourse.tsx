import styled from "styled-components";

const Wrapper = styled.section`
  border: 1px solid ${(props) => props.theme.lightGrayTextColor};
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 30px;
  padding: 130px;

  img {
    width: 50%;
  }

  h1 {
    color: ${(props) => props.theme.textColor};
    font-size: 36px;
    font-weight: 900;
  }

  p {
    color: #667ba3;
    font-size: 17px;
  }
`;

const EmptyCourse = () => {
  return (
    <Wrapper>
      <img
        src="	https://urclass.codestates.com/static/media/emptyContentCover.55f563d7.png"
        alt=""
      />
      <h1>코스가 비어있습니다.</h1>
      <p>자세한 코스 정보는 마이페이지를 확인해주세요.</p>
    </Wrapper>
  );
};

export default EmptyCourse;
