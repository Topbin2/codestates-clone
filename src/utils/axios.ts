import axios from "axios";

export const custom = axios.create({
  baseURL: "https://jsonplaceholder.typicode.com",
});
