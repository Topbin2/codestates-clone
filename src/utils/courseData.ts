export const courseData = [
  {
    id: 1,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Section 1",
    body: "웹 프론트엔드 개발의 기초지식을 기반으로 스스로 단순한 Web App을 만들 수 있다.",
  },
  {
    id: 2,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Section 2",
    body: "서버와 통신이 가능한 구조적인 Web App을 만들 수 있다.",
  },
  {
    id: 3,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Section 3",
    body: "사용자 친화적이고 안전한 Web App을 만들 수 있다.",
  },
  {
    id: 4,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Section 4",
    body: "사람과 기계가 모두 쉽고 빠르게 접근 가능한 Web App을 만들 수 있다.",
  },
  {
    id: 5,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Project",
    body: "First Project, Final Project",
  },
  {
    id: 6,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Wiki & Survey",
    body: "SEB FE 부트캠프에서 활용되고 있는 Wiki & Survey",
  },
];
