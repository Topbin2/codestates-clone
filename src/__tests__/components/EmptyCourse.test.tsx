import EmptyCourse from "../../components/EmptyCourse";
import { render, screen } from "../../utils/test-utils";
import renderer from "react-test-renderer";

describe("EmptyCourse Component", () => {
  it("snapshot", () => {
    const component = renderer.create(<EmptyCourse />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("img 태그는 해당 속성을 포함한다.", () => {
    render(<EmptyCourse />);
    const img = screen.getByAltText("빈 코스");
    expect(img).toHaveAttribute(
      "src",
      "https://urclass.codestates.com/static/media/emptyContentCover.55f563d7.png"
    );
  });

  it("h1 태그는 해당 텍스트를 포함한다.", () => {
    render(<EmptyCourse />);
    const h1 = screen.getByRole("heading");
    expect(h1).toHaveTextContent("코스가 비어있습니다.");
  });

  it("p 태그는 해당 텍스트를 포함한다.", () => {
    const { container } = render(<EmptyCourse />);
    const p = container.querySelector("p");
    expect(p).toHaveTextContent(
      "자세한 코스 정보는 마이페이지를 확인해주세요."
    );
  });
});
