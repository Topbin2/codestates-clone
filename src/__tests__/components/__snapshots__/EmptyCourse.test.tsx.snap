// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCourse Component snapshot 1`] = `
<section
  className="sc-bczRLJ dGymdG"
>
  <img
    alt="빈 코스"
    src="https://urclass.codestates.com/static/media/img-empty-content.55f563d7.png"
  />
  <h1>
    코스가 비어있습니다.
  </h1>
  <p>
    자세한 코스 정보는 마이페이지를 확인해주세요.
  </p>
</section>
`;
